docs(request-validator) Add docs for JSON Schema Draft 4 validation (#2093)
Kong/docs.konghq.com: app/_hub/kong-inc/request-validator/index.md

desc: Validates requests before they reach the upstream service
description: |

>   Validate requests before they reach their upstream service. Supports validating

Reviewer: ```suggestion
  Validate requests before they reach their upstream Service. Supports validating
```

Service always initial-capped

  the schema of the body and the parameters of the request using either Kong's own

>   schema validator (body only) or a JSON Schema Draft 4 compliant validator.

Reviewer: ```suggestion
  schema validator (body only) or a JSON Schema Draft 4-compliant validator.
```

    - name: allowed_content_types
      required: false
      default: "application/json"
      value_in_examples:
      description: |

>         List of allowed content types. <br>*Note: Body validation is only

Reviewer: ```suggestion
        List of allowed content types. <br>**Note:** Body validation is only
```

>         done for `application/json` and skipped for any other allowed content types.*

Reviewer: ```suggestion
        done for `application/json` and skipped for any other allowed content types.
```

    - name: version
      required: true
      default: "kong"
      value_in_examples:
      description: |
        What validator to use. Supported values are `kong` (default) for using Kong's own schema

>         validator, or `draft4` for using a JSON Schema Draft 4 compliant validator.

Reviewer: ```suggestion
        validator, or `draft4` for using a JSON Schema Draft 4-compliant validator.
```

- name: request-validator
  service: {service}
  config:
    version: kong
    body_schema:
      name:
        type: string
        required: true
```
{% endtabs %} 

> In this example, the request body data would have to be a valid JSON and

Reviewer: ```suggestion
In this example, the request body data would have to be valid JSON and
```

conform to the schema specified in `body_schema` - i.e., it would be required
to contain a `name` field only, which needs to be a string.

In case the validation fails, a `400 Bad Request` will be returned as the response.

### Schema Definition

> *For using the JSON Schema Draft 4 compliant validator please see the [JSON Schema website](

Reviewer: ```suggestion
*For using the JSON Schema Draft 4-compliant validator, see the [JSON Schema website](
```

https://json-schema.org/) for details on the format and examples. The rest of
this paragraph will explain the Kong schema.*

The `config.body_schema` field expects a JSON array with the definition of each

> field expected to be in the request body, for example:

Reviewer: ```suggestion
field expected to be in the request body; for example:
```
existing punctuation was correct

| `timestamp` | Integers | True if the field value is a valid timestamp |
| `uuid`| Strings | True if the string is a valud UUID |

> **Note**: See [here][lua-patterns] to learn about Lua patterns.

Reviewer: ```suggestion
**Note**: To learn more, see [Lua patterns][lua-patterns].
```
Avoid "here". State subject before an xref.

### Further References

> See the Kong docs on storing custom entities [here][schema-docs].

Reviewer: ```suggestion
For more information, see the Kong docs on [storing custom entities][schema-docs].
```
